apps/user-service/src/auth/reset-password.controller.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Controller,
+  Body,
+  Post,
+  HttpStatus,
+  BadRequestException,
+  Param, Put
+} from "@nestjs/common";
+import { ResetPasswordService } from './reset-password.service';
+import { ApiOperation } from '@nestjs/swagger';
+import { JsonApiResponse } from '@terramatch-microservices/common/decorators';
+import { buildJsonApi, JsonApiDocument } from '@terramatch-microservices/common/util';
+import { ApiException } from '@nanogiants/nestjs-swagger-api-exception-decorator';
+import { ResetPasswordRequest } from "./dto/reset-password-request.dto";
+import { ResetPasswordDto } from './dto/reset-password.dto';
+import { NoBearerAuth } from '@terramatch-microservices/common/guards';
+import { ResetPasswordResponseDto } from "./dto/reset-password-response.dto";
+
+@Controller('auth/v3/passwordResets')
+export class ResetPasswordController {
+  constructor(private readonly resetPasswordService: ResetPasswordService) {}
+
+  @Post()
+  @NoBearerAuth
+  @ApiOperation({
+    operationId: 'requestPasswordReset',
+    description: 'Send password reset email with a token',
+  })
+  @JsonApiResponse({ status: HttpStatus.CREATED, data: { type: ResetPasswordResponseDto } })
+  @ApiException(() => BadRequestException, { description: 'Invalid request or email.' })
+  async requestReset(@Body() { emailAddress, callbackUrl }: ResetPasswordRequest): Promise<JsonApiDocument> {
+    const { email, userId } = await this.resetPasswordService.sendResetPasswordEmail(emailAddress, callbackUrl);
+    return buildJsonApi()
+      .addData(`${userId}`, new ResetPasswordResponseDto({ emailAddress: email }))
```

Comment:
Please use UUID here (and update the type on the DTO decorator). We should be favoring UUIDs for all API surface areas as much as possible. 

I realize the login controller is using numerical ID. It doesn't actually matter in that case, and I should probably change (and the user find endpoint; while only used for `me` right now, it does accept a numerical ID for future admin use that I want to switch to UUID). The recently added user update endpoint does use UUID though, and all other endpoints that exist today and refer to a real resource use UUID.